xtp/src/libxtp/pmlocalization.cc

```diff
@@ -74,7 +78,90 @@ void PMLocalization::computePML(Orbitals &orbitals) {
   }
   XTP_LOG(Log::error, log_) << TimeStamp() << " Orbitals localized after "
                             << iteration + 1 << " iterations" << std::flush;
-  orbitals.setPMLocalizedOrbital(occupied_orbitals);
+
+  // check if localized orbitals orthonormal, if nor warn
+  check_orthonormality();
+
+  // determine the energies of the localized orbitals
+  Eigen::VectorXd energies = calculate_lmo_energies(orbitals);
+
+  // sort the LMOS according to energies
+  auto [LMOS, LMOS_energies] = sort_lmos(energies);
+  // print info
+  for (Index i = 0; i < LMOS_energies.size(); i++) {
+    XTP_LOG(Log::error, log_)
+        << (boost::format(" LMO index = %1$4i Energy = %2$10.5f Hartree ") %
+            (i) % LMOS_energies(i))
+               .str()
+        << std::flush;
+  }
+  orbitals.setLMOs(LMOS);
+  orbitals.setLMOs_energies(LMOS_energies);
+}
+
+// sort the LMOs according to their energy
+std::pair<Eigen::MatrixXd, Eigen::VectorXd> PMLocalization::sort_lmos(
+    const Eigen::VectorXd &energies) {
+  Eigen::VectorXd LMOS_energies(energies.size());
+  Eigen::MatrixXd LMOS(localized_orbitals_.rows(), localized_orbitals_.cols());
+
+  // sort the LMOs according to energy
+  std::vector<std::pair<double, int> > vp;
+
+  // Inserting element in pair vector
+  // to keep track of previous indexes
+  for (int i = 0; i < energies.size(); ++i) {
+    vp.push_back(std::make_pair(energies(i), i));
+  }
+  // Sorting pair vector
+  std::sort(vp.begin(), vp.end());
+
+  for (Index i = 0; i < energies.size(); i++) {
+    LMOS_energies(i) = vp[i].first;
+    LMOS.col(i) = localized_orbitals_.col(vp[i].second);
+  }
+
+  return {LMOS, LMOS_energies};
+}
+
+// calculate energies of LMOs
+Eigen::VectorXd PMLocalization::calculate_lmo_energies(Orbitals &orbitals) {
+  Eigen::MatrixXd h = overlap_ * orbitals.MOs().eigenvectors() *
+                      orbitals.MOs().eigenvalues().asDiagonal() *
+                      orbitals.MOs().eigenvectors().transpose() * overlap_;
+  // Eigen::VectorXd energies =
+  return (localized_orbitals_.transpose() * h * localized_orbitals_).diagonal();
```

Comment:
if this is a performance bottleneck you can make it cheaper by only calculating the diagonal elements.
```suggestion
  return (localized_orbitals_.transpose() * h).cwiseProduct(localized_orbitals_.transpose());
```